openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -1,14 +1,37 @@
 from rest_framework import serializers
+from django.conf import settings
 from django.contrib.auth.models import User
 from openedx.core.djangoapps.user_api.accounts import NAME_MIN_LENGTH
 from openedx.core.djangoapps.user_api.serializers import ReadOnlyFieldsSerializerMixin
 
-from student.models import UserProfile
+from student.models import UserProfile, LanguageProficiency
 from .helpers import get_profile_image_url_for_user, PROFILE_IMAGE_SIZES_MAP
 
 PROFILE_IMAGE_KEY_PREFIX = 'image_url'
 
 
+class LanguageProficiencySerializer(serializers.ModelSerializer):
```

Comment:
The main difference is that HyperlinkedModelSerializers serialize their identity as urls instead of ids. But this is being embedded anyways (e.g. language proficiency is not a separate URL), and you've overridden the id behavior to use the language code -- so there's no need to use a HyperlinkedModelSerializer.